src/main/kotlin/com/wolt/arrow/detekt/rules/NoEffectScopeBindableValueAsStatement.kt

```diff
@@ -105,13 +110,27 @@ class NoEffectScopeBindableValueAsStatement(config: Config) : Rule(config) {
                 .getQualifiedExpressionForReceiver()
                 ?.selectorExpression
 
-        private fun isBindable(type: KotlinType): Boolean =
-            type
-                .constructor
-                .declarationDescriptor
-                ?.fqNameSafe
-                ?.let { it in BindableFqNames }
-                ?: false
+        private fun isBindable(type: KotlinType): Boolean {
+            val isBindableFqNames =
+                type
+                    .constructor
+                    .declarationDescriptor
+                    ?.fqNameSafe
+                    ?.let { it in BindableFqNames } ?: false
+
+            val isPackage = (type.getAbbreviatedType()
+                ?.abbreviation
+                ?.constructor
+                ?.declarationDescriptor
+                ?.containingDeclaration as? PackageFragmentDescriptorImpl
+              )?.fqName?.asString() == "arrow.core.raise"
+
+            val isRaiseEffect =
+                (type.getAbbreviatedType()?.abbreviation?.constructor?.declarationDescriptor?.name?.asString() == "EagerEffect") ||
+                  (type.getAbbreviatedType()?.abbreviation?.constructor?.declarationDescriptor?.name?.asString() == "Effect")
+
+            return isBindableFqNames || (isPackage && isRaiseEffect)
```

Comment:
```suggestion
            val isRaiseExtensionFunction = type.annotations.hasAnnotation(FqName("kotlin.ExtensionFunctionType")) &&
                type
                    .arguments
                    .firstOrNull()
                    ?.type
                    ?.constructor
                    ?.declarationDescriptor
                    ?.fqNameSafe == FqName("arrow.core.raise.Raise")

            return isBindableFqNames || isRaiseExtensionFunction
```
Would it be safe to assume that any extension function on `Raise` is bindable? That should fix the problem with typealiases as well

Reply:
No, actually the contrary. An extension function on `Raise` will typically not return a _wrapper_.

I.e.

```kotlin
fun Raise<String>.example(): Int = raise("failure")

val either: Either<String, Int> = either { example() }
```

If it does return a wrapper, it should be caught by one of the other predicate in `isBindable`. I was trying to detect `Effect` & `EagerEffect` which now are respectively `typealias Effect<E, A> = suspend Raise<E>.() -> A` and `typealias EagerEffect<E, A> = Raise<E>.() -> A`.

Since they're not types anymore, they can not as easily be _detekted_. So what I was doing in the predicate was trying to look for _builders_ of `Effect` (and `EagerEffect`).

```kotlin
import arrow.core.raise.Effect
import arrow.core.raise.effect
                 
fun test(): Effect<Throwable, Int> = effect {
  effect <Throwable, Int> { 1 }
  1
}
```

The problem I faced was that while `EagerEffect` shows up as an `abbreviation` which I assumed was compiler language for `typealias`, that is not the case for `Effect` which simply turns up as `SuspendFunction` 😟

Not sure if there is other ways of detecting this, for example:
 - Checking if `arrow.core.raise.effect` was invoked inside a `Raise<E>.() -> A` builder and not _bound_.

Reply:
> No, actually the contrary. An extension function on Raise will typically not return a wrapper.

Oh yeah, I think I worded it wrong, would it be safe to assuming that any expression with type `Raise<Error>.() -> A` (regardless of `suspend`) is bindable (inside suitable scope)?

But that's an excellent example with `fun Raise<String>.example()`! Seems like we'd need to support it as well because body of such extension function is a "bindable scope" as well. What do you think about something like [this](https://github.com/nomisRev/arrow-detekt-rules/compare/update-to-1.2.0-raise...woltapp:arrow-detekt-rules:detekt-raise-extension-fns?expand=1), does it look like it would cover all use cases? 🤔 
All of the tests from `NoRaiseBindableValueAsStatementTest` pass in this branch except for Validated test due to using `2.0.0-SNAPSHOT`

Reply:
> What do you think about something like [this](https://github.com/nomisRev/arrow-detekt-rules/compare/update-to-1.2.0-raise...woltapp:arrow-detekt-rules:detekt-raise-extension-fns?expand=1), does it look like it would cover all use cases? 🤔


That looks awesome!! 😍 Your changes looks much cleaner as well 🥳
How do you want to proceed? You should be able to push your changes directly to this PR, and then we can merge it ☺️ These changes will be released before KotlinConf 😉

Reply:
Great, thank you! Let's merge it then :)

Reply:
Released version `0.2.0` with these changes